src/Microsoft.Azure.WebJobs.ServiceBus/EventHubs/EventHubTriggerInputBindingStrategy.cs

```diff
@@ -63,9 +70,34 @@ public EventData[] BindMultiple(EventHubTriggerInput value, ValueBindingContext
                 throw new ArgumentNullException("value");
             }
 
-            Dictionary<string, object> bindingData = new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase);
-            bindingData[DataContractPartitionContext] = value.Context;
+            var bindingData = new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase);
+            SafeAddValue(() => bindingData.Add(nameof(value.PartitionContext), value.PartitionContext));
+
+            if (value.IsSingleDispatch)
+            {
+                EventData eventData = value.GetSingleEventData();
+                SafeAddValue(() => bindingData.Add(nameof(eventData.PartitionKey), eventData.PartitionKey));
+                SafeAddValue(() => bindingData.Add(nameof(eventData.Offset), eventData.Offset));
+                SafeAddValue(() => bindingData.Add(nameof(eventData.SequenceNumber), eventData.SequenceNumber));
+                SafeAddValue(() => bindingData.Add(nameof(eventData.EnqueuedTimeUtc), eventData.EnqueuedTimeUtc));
+                SafeAddValue(() => bindingData.Add(nameof(eventData.Properties), eventData.Properties));
+                SafeAddValue(() => bindingData.Add(nameof(eventData.SystemProperties), eventData.SystemProperties));
+            }
```

Comment:
I've updated the code such that it will use arrays in the multiple dispatch case